kuma/search/views.py

```diff
@@ -115,11 +116,14 @@ class SearchRedirectView(RedirectView):
 
     def get_redirect_url(self, *args, **kwargs):
         query_string = self.request.META.get("QUERY_STRING")
-        url = reverse_lazy(
-            "api.v1.search", kwargs={"locale": self.request.LANGUAGE_CODE}
-        )
-        if query_string:
-            url += "?" + query_string
+        url = reverse_lazy("api.v1.search")
+        qs = parse_qs(query_string)
+        # If you used `/en-Us/search.json` you can skip the `?locale=`
+        # because the default locale in `/api/v1/search` is `en-US`.
+        if self.request.LANGUAGE_CODE.lower() != settings.LANGUAGE_CODE.lower():
```

Comment:
This also ensures that the locale in the URL (the request) takes precedent if a locale is also specified in the query string and they're different, which seems fine to me.